### PR TITLE
pass tests

### PR DESCRIPTION
**Pull request by ashleygwilliams**
READ, UPDATE, and DELETE functionality totally works, the tests don't pass tho

CREATE still has issue #77

### CODE REVIEW

**Comment by ashleygwilliams** [comment]
currently at 20 of 70 failing

**Comment by ashleygwilliams** [comment]
17 of 70

File: api/classes/base_controller.js [hunk 1]
```diff
@@ -41,37 +41,25 @@ BaseController.prototype.create = function(req, reply) {
...
+      throw Boom.create(404);
     }
+    return record
+      .save(record.changed, { patch: true, method: 'update' });
   })
-  .save(this.data(req), {
-    method: 'update',
-    patch: 'true'
-  });
-  var updated_state = this.Model.query({
```
Comment (sedge) [inline api/classes/base_controller.js]: I have no idea how this worked before. All this stuff happening in parallel must have been a nightmare to debug.

File: api/modules/files/controller.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ controller.data = function(req) {
     buffer: req.payload.buffer
   };
   if (req.params.id) {
-    data.id = req.params.id;
+    data.id = parseInt(req.params.id);
```
Comment (sedge) [inline api/modules/files/controller.js]: Strange that you have to do this. It's an odd place for this to be forced. Why not add param validation with Joi?

File: api/modules/files/routes/update.js [hunk 3]
```diff
@@ -8,7 +10,11 @@ module.exports = [{
     handler: controller.update.bind(controller),
     description: 'Update a single file object based on `id`.',
     validate: {
-      payload: schema
+      payload: schema,
+      params: {
```
Comment (sedge) [inline api/modules/files/routes/update.js]: Huh. To my point above, why isn't this enough?

Comment (sedge) [inline api/modules/files/routes/update.js]: Could add to `create` too?

**Comment by ashleygwilliams** [comment]
16 of 70

**Comment by sedge** [comment]
Closed in favour of #87

**Pull request closed without merging**